### PR TITLE
Basic Support for LIvestatus based Servers

### PR DESCRIPTION
**Pull request by fragfutter**
Basic support for a livestatus based server. 

Developed against icinga2. 

Implements monitoring, recheck, acknowledge

Missing open_monitor, open_monitor_webpage, set_downtime, set_submit_check_result

### CODE REVIEW

**Comment by HenriWahl** [comment]
Thanks for contributing to the qt branch! Just one question: what are the changes in QUI/__init__.py for? At a first glance I do not understand them.

File: Nagstamon/Nagstamon/QUI/__init__.py [hunk 1]
```diff
@@ -327,7 +327,7 @@ def show_at_cursor(self):
         # get cursor coordinates and decrease them to show menu under mouse pointer
         x = QCursor.pos().x() - 10
         y = QCursor.pos().y() - 10
-        self.exec(QPoint(x, y))
+        self.exec(QPoint(x, y))  # noqa
```
Comment (fragfutter) [inline Nagstamon/Nagstamon/QUI/__init__.py]: silence pep8 checks

File: Nagstamon/Nagstamon/QUI/__init__.py [hunk 2]
```diff
@@ -4679,6 +4680,7 @@ def exit():
 systrayicon = SystemTrayIcon()
 
 # combined statusbar/status window
+statuswindow = None
```
Comment (fragfutter) [inline Nagstamon/Nagstamon/QUI/__init__.py]: make sure variable is known

File: Nagstamon/Nagstamon/QUI/__init__.py [hunk 3]
```diff
@@ -2743,6 +2743,7 @@ def get_status(self):
 
                 # if failures have gone and nobody took notice switch notification off again
                 if len([k for k,v in self.server.events_history.items() if v == True]) == 0 and\
+                        statuswindow and \
```
Comment (fragfutter) [inline Nagstamon/Nagstamon/QUI/__init__.py]: check that a statuswindow exists before accessing it's properties

**Comment by fragfutter** [comment]
it fixes a race condition i had at some point. During the creation of statuswindow it tried to check if some properties on statuswindow where set. I made sure that the variable is defined (as None), and before accessing properties check if it is initialized.

the noqa comment is only to silence pep8. I left comments

https://github.com/HenriWahl/Nagstamon/pull/180/files

**Pull request merged**